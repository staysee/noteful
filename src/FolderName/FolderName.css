.FolderName__name {
    color: white;
    writing-mode: vertical-rl;
    text-orientation: upright;
    margin: 20px auto;
    font-size: 30px;
}

.goBack-button {
    border-radius: 50%;
    height: 70px;
    width: 70px;
    border: 0;
    margin: 0 auto;
    display: block;
    margin-top: 30px;
    background-color: #1b8179;
    color: white;
    font-weight: bold;
}

.goBack-button:hover {
    cursor: pointer;
    background-color: #FFB72F;
}