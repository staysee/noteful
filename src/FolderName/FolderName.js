import React from 'react'
import './FolderName.css'

function FolderName(props){
    return (
        <div className="FolderName">
            <button
                className="goBack-button"
                onClick={() => props.history.goBack()}
            >
                Back
            </button>

            {props.folder && (
                <h3 className="FolderName__name">
                    {props.folder.name}
                </h3>
            )}

        </div>
    )
}

FolderName.defaultProps = {
    history: {
        goBack: () => {}
    }
}

export default FolderName