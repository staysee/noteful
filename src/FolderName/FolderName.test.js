import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FolderName from './FolderName'

describe(`FolderName component`, () => {
    it('renders a .FolderName by default', () => {
        const wrapper = shallow(<FolderName />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    //enzyme doesn't support React.createContext
    it.skip('renders a h3 with folder name when in props', () => {
        const props = {
            match: {
                params: {
                    noteId: 'test-note-id'
                }
            }
        }

        const context = {
            notes: [{ id: 'test-note-id', folderId: 'test-folder-id' }],
            folders: [{ id: 'test-folder-id', name: 'Important' }]
          }
    
        const h3 = shallow(<FolderName {...props} />, context)
            .find('.FolderName__name')
        expect(toJson(h3)).toMatchSnapshot()
    })
})