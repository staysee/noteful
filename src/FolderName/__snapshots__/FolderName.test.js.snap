// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderName component renders a .FolderName by default 1`] = `
<div
  className="FolderName"
>
  <button
    className="goBack-button"
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;

exports[`FolderName component renders a h3 with folder name when in props 1`] = `null`;
